### PR TITLE
update ROOT easyblock to support recent versions that require using CMake, add sanity check, clean up/enhance make_module*

### PR DESCRIPTION
**Pull request by boegel**
required for https://github.com/easybuilders/easybuild-easyconfigs/pull/5081

### CODE REVIEW

File: easybuild/easyblocks/r/root.py [hunk 1]
```diff
@@ -28,43 +28,106 @@
...
+                self.cfg.update('configopts', '-Dbuiltin_fftw3=OFF -DFFTW_DIR=%s' % fftw_root)
+
+            gsl_root = get_software_root('GSL')
+            if gsl_root:
+                self.cfg.update('configopts', '-DGSL_DIR=%s' % gsl_root)
+
+            mesa_root = get_software_root('Mesa')
+            if mesa_root:
+                self.cfg.update('configopts', '-DDOPENGL_INCLUDE_DIR=%s' % os.path.join(mesa_root, 'include'))
+                self.cfg.update('configopts', '-DDOPENGL_INCLUDE_DIR=%s' % os.path.join(mesa_root, 'lib', 'libGL.so'))
```
Comment (wpoely86) [inline easybuild/easyblocks/r/root.py]: LIB?

Comment (boegel) [reply easybuild/easyblocks/r/root.py]: Should be `-DOPENGL_gl_LIBRARY`... Apparently ROOT is able to find the library regardless, but this is clearly wrong, I'll fix it.

File: easybuild/easyblocks/i/icc.py [hunk 2]
```diff
@@ -44,6 +44,9 @@
 from easybuild.tools.run import run_cmd
 from easybuild.tools.systemtools import get_shared_lib_ext
 
+# location system header files on Debian/Ubuntu that Intel compilers must consider
```
Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py]: only intel, not gcc?

Comment (boegel) [reply easybuild/easyblocks/i/icc.py]: this is not supposed to be in here, urgh, will remove...

**Pull request merged**